front/components/ConnectorPermissionsModal.tsx

```diff
@@ -28,71 +46,320 @@ const PERMISSIONS_EDITABLE_CONNECTOR_TYPES: Set<ConnectorProvider> = new Set([
+                break the existing data structure in Dust and Assistants using
+                them.
+              </div>
+
+              <div className="pl-4 text-sm text-amber-800">
+                Read our{" "}
+                <a
+                  href="https://docs.dust.tt/docs/google-drive-connection"
+                  className="text-blue-600"
+                >
+                  Playbook
+                </a>
+                .
+              </div>
+            </div>
+            <div className="flex items-center justify-center">
+              <Button
+                label={"Edit Permissions"}
+                icon={LockIcon}
+                variant="primaryWarning"
+                onClick={() => {
+                  setShowConfirmDialog(true);
+                }}
+              />
+            </div>
+          </div>
+        )}
+        {isDataSourceOwner && (
+          <div className="flex items-center justify-center">
+            <Button
+              label={"Edit Permissions"}
+              icon={LockIcon}
+              variant="primaryWarning"
+              onClick={() => {
+                setShowConfirmDialog(true);
+              }}
+            />
+          </div>
+        )}
+        <Dialog
+          title="Are you sure?"
+          isOpen={showConfirmDialog}
+          onCancel={() => setShowConfirmDialog(false)}
+          onValidate={() => {
+            void onEditPermissionsClick();
+            setShowConfirmDialog(false);
+          }}
+          validateVariant="primaryWarning"
+          cancelLabel="Cancel"
+          validateLabel="Continue"
+        >
+          The changes you are about to make may break existing{" "}
+          {connectorConfiguration.name} Data sources and the assistants using
+          them. Are you sure you want to continue?
+        </Dialog>
+      </>
+    </DataSourceManagementModal>
+  );
+}
+
 export function ConnectorPermissionsModal({
   owner,
   connector,
   dataSource,
   isOpen,
   onClose,
-  setShowEditionModal,
-  handleUpdatePermissions,
   plan,
   readOnly,
   isAdmin,
+  dustClientFacingUrl,
+  user,
+  onManageButtonClick,
 }: {
   owner: WorkspaceType;
   connector: ConnectorType;
   dataSource: DataSourceType;
   isOpen: boolean;
   onClose: () => void;
-  setShowEditionModal: (show: boolean) => void;
-  handleUpdatePermissions: (
-    connector: ConnectorType,
-    dataSource: DataSourceType
-  ) => Promise<void>;
   plan: PlanType;
+  dustClientFacingUrl: string;
   readOnly: boolean;
   isAdmin: boolean;
+  user: UserType;
+  onManageButtonClick?: () => void;
 }) {
   const { mutate } = useSWRConfig();
 
   const [updatedPermissionByInternalId, setUpdatedPermissionByInternalId] =
     useState<Record<string, ConnectorPermission>>({});
+  const [showEditionModal, setShowEditionModal] = useState(false);
+  const [showDataSelectionModal, setShowDataSelectionModal] = useState(false);
```

Comment:
can only one of theses be true at the same time ?
if so, what about a `const modalToShow:string = "" | "edition" | "selection";` to enforce the behavior ?